Subject: [PATCH] ppc: exec stack fix

execstack shows ppc kexec has an executable stack,
this leaves it vulnerable to buffer overflows.

Fix it by adding ASFLAGS --noexecstack

Tested on PowerMac G4 Macmini:
Without the patch:
dave@darkstar:~/kexec-tools$ execstack build/sbin/kexec
X build/sbin/kexec

With the patch:
dave@darkstar:~/kexec-tools$ execstack build/sbin/kexec
- build/sbin/kexec

Signed-off-by: Dave Young <dyoung@redhat.com>
Signed-off-by: Simon Horman <horms@verge.net.au>
---
 kexec/arch/ppc/Makefile |    2 ++
 1 file changed, 2 insertions(+)

Index: kexec-tools-2.0.3/kexec/arch/ppc/Makefile
===================================================================
--- kexec-tools-2.0.3.orig/kexec/arch/ppc/Makefile
+++ kexec-tools-2.0.3/kexec/arch/ppc/Makefile
@@ -23,6 +23,8 @@ CPPFLAGS+=-I$(srcdir)/kexec/arch/$(ARCH)
 
 ppc_KEXEC_SRCS += $(libfdt_SRCS)
 
+ASFLAGS	+= -Wa,--noexecstack
+
 dist += kexec/arch/ppc/Makefile $(ppc_KEXEC_SRCS)			\
 	kexec/arch/ppc/crashdump-powerpc.h kexec/arch/ppc/fixup_dtb.h	\
 	kexec/arch/ppc/kexec-ppc.h kexec/arch/ppc/ops.h			\
